lib/libc/minimal/source/stdlib/malloc.c

```diff
@@ -0,0 +1,124 @@
+#include <stdlib.h>
+#include <zephyr.h>
+#include <init.h>
+#include <errno.h>
+#include <misc/mempool.h>
+#include <string.h>
+
+#if (CONFIG_MALLOC_ARENA_SIZE > 0)
+K_MUTEX_DEFINE(malloc_mutex);
+SYS_MEM_POOL_DEFINE(z_malloc_mem_pool, &malloc_mutex,  64,
+		    CONFIG_MALLOC_ARENA_SIZE, 1, 4, .data);
```

Comment:
FWIW: 64 is a pretty chunky minium size.  The overhead for allowing smaller blocks isn't super high (amortized it's like 12-ish bytes per power of two level, I think?), so you could probably go lower if you wanted.  The absolute minimum is 8.

Reply:
Done.